### PR TITLE
[ubuntu24.04] remove usage of deprecated 'apt-key add'

### PR DESCRIPTION
**Pull request by cdesiniotis**

### CODE REVIEW

File: ubuntu24.04/Dockerfile [hunk 1]
```diff
@@ -64,8 +64,11 @@ RUN echo "TARGETARCH=$TARGETARCH"
 ADD install.sh /tmp
 
 # Fetch GPG keys for CUDA repo
-RUN apt-key del 7fa2af80 && OS_ARCH=${TARGETARCH/amd64/x86_64} && OS_ARCH=${OS_ARCH/arm64/sbsa} && \
-    apt-key adv --fetch-keys "https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH}/3bf863cc.pub"
+RUN apt-key del 7fa2af80 3bf863cc && \
```
Comment (tariq1890) [inline ubuntu24.04/Dockerfile:67]: Do we know if these GPG keys are already added to the container image?

If not, we could just remove it and add them when it becomes necessary

File: ubuntu24.04/Dockerfile [hunk 2]
```diff
@@ -64,8 +64,11 @@ RUN echo "TARGETARCH=$TARGETARCH"
...
+    apt-get update && apt-get install -y --no-install-recommends curl gpg && \
+    OS_ARCH=${TARGETARCH/amd64/x86_64} && OS_ARCH=${OS_ARCH/arm64/sbsa} && \
+    curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH}/3bf863cc.pub | gpg --dearmor -o /etc/apt/keyrings/cuda-3bf863cc.pub && \
+    echo "deb [signed-by=/etc/apt/keyrings/cuda-3bf863cc.pub] https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH} /" > /etc/apt/sources.list.d/cuda.list
```
Comment (tariq1890) [inline ubuntu24.04/Dockerfile:71]: Just a nit, but why not this? Just looks a tad better IMO
```suggestion
    echo "deb [signed-by=/etc/apt/keyrings/cuda.pub] https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH} /" > /etc/apt/sources.list.d/cuda.list
```

File: ubuntu24.04/Dockerfile (hunk 1 above)
Comment (cdesiniotis) [reply ubuntu24.04/Dockerfile:67]: 3bf863cc is included. 7fa2af80 is the old one and it is not in the ubuntu 24.04 image. I can omit 7fa2af80 from this command.

File: ubuntu24.04/Dockerfile (hunk 2 above)
Comment (cdesiniotis) [reply ubuntu24.04/Dockerfile]: Updated to just `cuda.pub`

File: ubuntu24.04/Dockerfile [hunk 3]
```diff
@@ -64,8 +64,12 @@ RUN echo "TARGETARCH=$TARGETARCH"
 ADD install.sh /tmp
 
 # Fetch GPG keys for CUDA repo
-RUN apt-key del 7fa2af80 && OS_ARCH=${TARGETARCH/amd64/x86_64} && OS_ARCH=${OS_ARCH/arm64/sbsa} && \
-    apt-key adv --fetch-keys "https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/${OS_ARCH}/3bf863cc.pub"
+RUN apt-key del 3bf863cc && \
```
Comment (tariq1890) [inline ubuntu24.04/Dockerfile:67]: Looks like this statement is running in the final layer, can we `apt purge` the curl package after performing this action?

Comment (tariq1890) [reply ubuntu24.04/Dockerfile:67]: Let's also add this line as it helps shrink the image size further

```
rm -rf /var/lib/apt/lists/*; \
```

**Review by tariq1890 — APPROVED** [review]

**Review by tariq1890 — APPROVED** [review]

**Pull request merged**